src/unstract/sdk/file_storage/fs_impl.py

```
@@ -0,0 +1,187 @@
+            raise FileOperationError(str(e))
+
+    def write(
+        self,
+        path: str,
+        mode: str,
+        encoding: str = Common.DEFAULT_ENCODING,
+        seek_position: int = 0,
+        data: Union[bytes, str] = "",
+    ) -> int:
+        """Write data in the file pointed to by the file-handle.
+
+        Args:
+            path (str): Path to the file to be opened
+            mode (str): Mode in whicg the file is to be opened. Usual options
+                        include r, rb, w and wb
+            encoding (str): Encoding type like utf-8 or utf-16
+            seek_position (int): Position to start writing from
+            data (Union[bytes, str]): Contents to be written
+
+        Returns:
+            int: Number of bytes that were successfully written to the file
+        """
+        try:
+            with self.fs.open(path=path, mode=mode, encoding=encoding) as file_handle:
+                return file_handle.write(data)
+        except Exception as e:
+            raise FileOperationError(str(e))
+
+    def seek(
+        self,
+        path: str,
+        location: int = 0,
+        position: FileSeekPosition = FileSeekPosition.START,
+    ) -> int:
+        """Place the file pointer to the mentioned location in the file
+        relative to the position.
+
+        Args:
+            path (str): path of the file
+            location (int): Nth byte position. To be understood in relation to
+            the arg "position"
+            position (FileSeekPosition): from start of file, current location
+            or end of file
+
+        Returns:
+            int: file pointer location after seeking to the mentioned position
+        """
+        try:
+            with self.fs.open(path=path, mode="rb") as file_handle:
+                return file_handle.seek(location, position)
+        except Exception as e:
+            raise FileOperationError(str(e))
+
+    def mkdir(self, path: str, create_parents: bool = True):
```

@gaya3-zipstack Can we add `is_exisit` param in the base method?

---

Silently ignored inside the implementation if file/path already exists